tests/core/test_processor.py

```diff
@@ -82,6 +90,39 @@ def loop():
         assert r
 
 
+async def mocked_parse(self, text, message_id=None, tracker=None):
+    """Mock parsing a text message and augment it with the slot
+    value from the tracker's state"""
+
+    return {
+        "intent": {"name": "", "confidence": 0.0},
+        "entities": [],
+        "text": text,
+        "requested_language": tracker.get_slot("requested_language")
+    }
+
+
+async def test_parsing_with_tracker():
+    message = UserMessage("lunch?")
+    tracker = DialogueStateTracker.from_dict("1", [], [Slot("requested_language")])
+
+    # we'll expect this value 'en' to be part of the result from the interpreter
+    tracker._set_slot("requested_language", "en")
+
+    endpoint = EndpointConfig("https://interpreter.com")
+    with aioresponses() as mocked:
+        mocked.post("https://interpreter.com/parse", repeat=True, status=200)
+
+        # mock the parse function with the one defined for this test
+        with patch.object(RasaNLUHttpInterpreter, "parse", mocked_parse):
+            interpreter = RasaNLUHttpInterpreter(endpoint=endpoint)
+            result = await MessageProcessor(interpreter, None, None, tracker, None)._parse_message(
+                message,
+                tracker
+            )
+            assert result["requested_language"] == "en"
```

Comment:
this test is not really verifying anything, apart from a successfully set slot. What should really be tested is the propagation of the tracker all the way through to the `_parse_message()` message

Reply:
Well, this assert doesn't check the presence of the slot, but the result of the parsing. The parsing will only include `requested_language` if the tracker is a) passed and b) has that slot set. Given other changes I've done, would it be acceptable?

Reply:
yes, so the change you made does test the propagation of the tracker 🙂